src/main/java/duke/maincomponents/Storage.java

```diff
@@ -0,0 +1,92 @@
+package main.java.duke.maincomponents;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.Writer;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import main.java.duke.exceptions.DukeException;
+import main.java.duke.exceptions.TaskException;
+import main.java.duke.task.Deadline;
+import main.java.duke.task.Event;
+import main.java.duke.task.Task;
+import main.java.duke.task.ToDo;
+
+
+public class Storage {
+    private String pathOfDataFile;
+
+    public Storage(String filePath, String folderName) throws DukeException {
+        pathOfDataFile = filePath;
+        try {
+            File folder = new File(folderName);
+            if (folder.mkdir()) {
+                //System.out.println("Folder created: " + folder.getName());
+            } else {
+                //System.out.println("Folder already exists.");
+            }
+            File dataFile = new File(filePath);
+            if (dataFile.createNewFile()) {
+                //System.out.println("File created: " + dataFile.getName());
+            } else {
+                //System.out.println("File already exists.");
+            }
+        } catch (IOException e) {
+            throw new DukeException("Error occured during file or folder creation");
+        }
+    }
```

Comment:
Instead of commenting out the codes that is not required, removing them will be better.

Reply:
i agree with u